lib/optimizely.rb

```diff
@@ -46,8 +47,22 @@ class Project
     #                 By default all exceptions will be suppressed.
     # @param user_profile_service - Optional component which provides methods to store and retreive user profiles.
     # @param skip_json_validation - Optional boolean param to skip JSON schema validation of the provided datafile.
-
-    def initialize(datafile, event_dispatcher = nil, logger = nil, error_handler = nil, skip_json_validation = false, user_profile_service = nil)
+    # @params sdk_key - Optional string uniquely identifying the datafile corresponding to project and environment combination.
+    #                   Must provide at least one of datafile or sdk_key.
+    # @param config_manager - Optional Responds to get_config.
+    # @param notification_center - Optional Instance of NotificationCenter.
+
+    def initialize(
+      datafile = nil,
+      event_dispatcher = nil,
+      logger = nil,
+      error_handler = nil,
+      skip_json_validation = false,
+      user_profile_service = nil,
+      sdk_key = nil,
```

Comment:
I am not in favor of adding all params. 
Either `sdk_key` or `config_manager` with `notification_center` should be enough. 
In case you add `sdk_key` then just add `config_manager` property to get access of PollingProjectConfigManager.

Reply:
I think if we want to make it a simpler initialization step, we can introduce the `OptimizelyFactory` class that we have in other SDKs that takes in less params and uses more defaults.

Reply:
Done